### PR TITLE
Update Smoke Testing docs to reference Product Collection block and Cart/Checkout templates

### PR DESCRIPTION
**Pull request by Aljullu**
## What & why

This PR updates the Smoke Testing docs so they reference the Product Collection block instead of the Products block. It also updates the step related to Cart & Checkout blocks to mention they need to be updated in the templates instead of the pages.

## Testing Instructions

1. Follow the `2. Create a page with the Products block, and filter blocks, setup to test that functionality in isolation. Using the columns block here too is a good idea to keep things organized` testing step and verify it works as expected.
2. Read the new testing steps and make sure they make sense.

* [x] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

### CODE REVIEW

**Review by opr — APPROVED** [review]
I tested the step and it works fine on my end! Thanks Albert.

File: docs/internal-developers/testing/smoke-testing.md [hunk 1]
```diff
@@ -167,22 +167,22 @@ Note: some blocks might fail to render because they are based on products having
...
-<div class="wp-block-query"><!-- wp:post-template {"__woocommerceNamespace":"woocommerce/product-query/product-template"} -->
-<!-- wp:woocommerce/product-image {"isDescendentOfQueryLoop":true,"style":{"spacing":{"margin":{"bottom":"0.75rem","top":"0"}}}} /-->
+<div class="wp-block-column" style="flex-basis:66.66%"><!-- wp:woocommerce/product-collection {"queryId":0,"query":{"perPage":9,"pages":0,"offset":0,"postType":"product","order":"asc","orderBy":"title","author":"","search":"","exclude":[],"sticky":"","inherit":false,"taxQuery":{},"parents":[],"isProductCollectionBlock":true,"woocommerceOnSale":false,"woocommerceStockStatus":["instock","outofstock","onbackorder"],"woocommerceAttributes":[],"woocommerceHandPickedProducts":[]},"tagName":"div","displayLayout":{"type":"flex","columns":3}} -->
```
Comment (kmanijak) [inline docs/internal-developers/testing/smoke-testing.md]: Let's just remove couple of unnecessary attributes:
- `author`, `sticky` and `parents` have been removed from the block and are not used anymore
- `queryId` will be automatically assigned when block is added in the Editor. In that case we can prevent unexpected consequences if `queryId` will be duplicated with other Product Collection block used during testing.

```suggestion
<div class="wp-block-column" style="flex-basis:66.66%"><!-- wp:woocommerce/product-collection {"query":{"perPage":9,"pages":0,"offset":0,"postType":"product","order":"asc","orderBy":"title"","search":"","exclude":[],"inherit":false,"taxQuery":{},"isProductCollectionBlock":true,"woocommerceOnSale":false,"woocommerceStockStatus":["instock","outofstock","onbackorder"],"woocommerceAttributes":[],"woocommerceHandPickedProducts":[]},"tagName":"div","displayLayout":{"type":"flex","columns":3}} -->
```

**Review by kmanijak — APPROVED** [review]
Nice improvement!

I left a small comment with attributes changes (they should be applied before merging!) but it's small enough that I'm pre-approving 🙌

File: docs/internal-developers/testing/smoke-testing.md (hunk 1 above)
Comment (Aljullu) [reply docs/internal-developers/testing/smoke-testing.md]: Done! Thank you both for your reviews! :raised_hands:

**Pull request merged**